### PR TITLE
Tapas v4 (tres)

### PR DESCRIPTION
**Pull request by NielsRogge**
Here we are again, opening a new PR based on the former (#8988) which had some Github issues. 
cc @LysandreJik

### CODE REVIEW

File: src/transformers/models/tapas/tokenization_tapas.py [hunk 1]
```diff
@@ -0,0 +1,2782 @@
...
+                            and (len(queries[0]) == 0 or isinstance(queries[0][0], str))
+                        )
+                    )
+                )
+            )
+        ), (
+            "queries input must of type `str` (single example), `List[str]` (batch or single pretokenized example) "
+            "or `List[List[str]]` (batch of pretokenized examples)."
+        )
+
```
Comment (LysandreJik) [inline src/transformers/models/tapas/tokenization_tapas.py:623]:
```suggestion
        # Input type checking for clearer error
        valid_query = False

        # Check that query has a valid type
        if queries is None or isinstance(queries, str):
            valid_query = True
        elif isinstance(queries, (list, tuple)):
            if len(queries) == 0 or isinstance(queries[0], str):
                valid_query = True

        if not valid_query:
            raise ValueError(
                "queries input must of type `str` (single example), `List[str]` (batch or single pretokenized example). "
            )
 ```

File: tests/test_modeling_tapas.py [hunk 2]
```diff
@@ -0,0 +1,1035 @@
...
+    def test_config(self):
+        self.config_tester.run_common_tests()
+
+    def test_model(self):
+        config_and_inputs = self.model_tester.prepare_config_and_inputs()
+        self.model_tester.create_and_check_model(*config_and_inputs)
+
+    # def test_for_masked_lm(self):
+    #     config_and_inputs = self.model_tester.prepare_config_and_inputs()
+    #     self.model_tester.create_and_check_for_masked_lm(*config_and_inputs)
```
Comment (LysandreJik) [inline tests/test_modeling_tapas.py:490]:
```suggestion
    def test_for_masked_lm(self):
        config_and_inputs = self.model_tester.prepare_config_and_inputs()
        self.model_tester.create_and_check_for_masked_lm(*config_and_inputs)
```

**Review by LysandreJik — APPROVED** [review]
After these changes are applied, LGTM. There are still some comments leftover from the previous PR. Do you want me to take care of it @NielsRogge?

**Comment by LysandreJik** [comment]
Don't worry about the TF tests, these are because of TF2.4 which are fixed on `master`.

**Comment by NielsRogge** [comment]
The conversion script currently includes a line in which I'm importing a local `vocab.txt`, I know this should be removed in the future.

**Review by sgugger — APPROVED** [review]

**Review by LysandreJik — APPROVED** [review]
Absolutely incredible work, thank you @NielsRogge for your patience and for your impressive work. Merging!

**Pull request merged**